### ISSUE TITLE
Subtracks, 2.0 exercises, tags

### ISSUE THREAD

**Issue by canweriotnow** [exercism/x-common#71]
Related to this [gitter convo](https://gitter.im/exercism/support?at=552fb77508c7a53a4a1d835a)

Also maybe related to #63 

I've been thinking about the possibility of interest tracks or subtracks:

**Example Case:** There's a lot of interest in logic programming and one of the popular things I see people doing is working through the exercises in [The Reasoned Schemer](http://mitpress.mit.edu/books/reasoned-schemer) in Clojure [core.logic](https://github.com/clojure/core.logic). Now that we have a [Scheme track](https://github.com/exercism/xscheme), making those exercises available in appropriate languages, with appropriate lib deps, would be a great way to work through them with others.

I also like the idea of iterative approaches to problems in the above-cited gitter convo. I think this might be orthogonal to subtracks, but I figured this issue would be a good place to tease out those distinctions.

As for #63 - right after the discussion about subtracks, it occurred to me that a tagging system might be a really useful thing. Classification like focus/practice/challenge is one area, but tags could also cover topical areas problems address, like string manipulation, list comprehension, iteration/recursion, etc.

Tags could also cover exercises that fall outside CS instruction justification; e.g., I was thinking about how differently I would implement ROT13 in Ruby, Scheme, C, etc. ROT13 is of course, at this point, a historical curiosity, but I think there are probably other things that would make for an interesting exercise with odd trivia in the README, e.g.:

```
Back in the olden days of USENET, it was considered courteous to obfuscate potentially 
offensive posts with ROT13 "encryption", which simply rotated characters 13 places so 
A becomes N, B becomes O, C becomes P, etc.

Implement ROT13 cipher for alphabetic characters, preserving case.
```

It's not useful, but could involve character encoding, modulo arithmetic, etc., depending on the implementation. Things like this could be tagged `historical` or `curiosity`.

Anyhow, just a few ideas on evolving both the granularity and scope of exercises on exercism.

Thoughts?

**Comment by kytrinyx** [comment]
+1000

I wonder if there would be copyright issues with The Reasoned Schemer exercises. It seems likely.

@pminten has had a lot of really interesting ideas in terms of expanding the scope and granularity of the exercises on exercism, I'll go find some of the links and paste them back here.

I feel like there are grounds for a very interesting discussion and making a plan for implementing a lot of these things.

**Comment by kytrinyx** [comment]
Here are some of the discussions that I was thinking of:

- https://github.com/exercism/x-common/issues/63
- https://github.com/exercism/exercism.io/issues/2262
- https://github.com/exercism/exercism.io/issues/2263
- https://github.com/exercism/exercism.io/issues/2268
- https://github.com/exercism/exercism.io/issues/2343

**Comment by Dog** [comment]
I put together a small problem to show what I was thinking regarding a problem that could be expanded upon. A couple ideas for expansion were changing requirements, expanding the feature set, a second customer, etc. I hadn't read any of pminten's ideas when writing the gist. 

I was thinking you could give a story for a bit of engagement. Having a customer also means that we can do things like suggest what may be best for them in nitpicking. The readme could then focus on what is going on in the problem conceptually.

https://gist.github.com/Dog/a257729d6c2e99b6685a

**Comment by canweriotnow** [comment]
Hmm, the copyright issue is interesting, and one I hadn't considered... I know a lot of people have reproduced the exercises as test cases in Scheme and Clojure and published their work on Github, but I guess a useful README would contain more of the original source material or at least a paraphrase... I'll do some asking around, I think I know someone who knows one or both of the authors and maybe we could get their blessing... one would think presenting the exercises in a forum like exercism would have a positive effect on book sales, but publishing is a wild and wacky world. MIT Press tends to be pretty liberal so I like our chances. 

I don't wanna sidetrack the discussion with that one case, but I'd be lying if I said there weren't other books/resources I'd like to borrow from, and probably not all of them are open-licensed.

**Comment by kytrinyx** [comment]
Authors are often generous... publishers less so, but you're right that MIT Press might give us better chances than other ones.

> I'd be lying if I said there weren't other books/resources I'd like to borrow from, and probably not all of them are open-licensed.

Yeah, in this vein I talked to Tom Stuart about his lovely "understanding computation" book.

**Comment by kytrinyx** [comment]
We are doing a complete redesign from the core principles and questions up. Today we published the first set of documents that explore questions around enhancing and supporting people's motivation to learn and practice on Exercism: https://github.com/exercism/docs/blob/master/about/conception/README.md

In particular, this issue is relevant to the section on "expectancies of success" and the related document on progression which explores a new take on how to structure exercises... using a variation on the idea of subtracks.

We have an issue open requesting thoughts and feedback on these design documents: https://github.com/exercism/discussions/issues/154

I'm going to close this issue as there has not been any new movement in it for quite some time, but we've tagged it so that we can find it when we start exploring the classification part of the redesign.

Thanks, everyone, for helping think this through.
